Events/ProxyFilterControllerEvent.php

```diff
@@ -5,7 +5,7 @@
 use Symfony\Component\HttpKernel\Event\ControllerEvent;
 use Symfony\Component\HttpKernel\Event\FilterControllerEvent as LegacyEvent;
 
-if (!class_exists('Symfony\\Component\HttpKernel\\Event\\ControllerEvent')) {
+if (!class_exists('Symfony\\Component\\HttpKernel\\Event\\ControllerEvent')) {
```

Comment:
Just looking at this, but i guess that this fix (and the other way the same) is probably the real bug, and the changes to remove the typehints are probably not needed?

Reply:
Nah I did see and fix that first, type hinting still fucks it up

Reply:
It makes sense if you think about it.
Cause we are requesting the more specific subclass it throws an error cause you give it the parent.
If we'd request the SF ControllerEvent and feed it with a ProxyFilterControllerEvent it would work fine

Reply:
yes, i see what you mean